main.go

```diff
@@ -58,6 +58,7 @@ func main() {
 	defaultTokenExpiration := flag.Duration("token-expiration", webhooks.DefaultTokenExpirationDefault, "The token expiration")
 	defaultRegion := flag.String("gcp-default-region", "", "If set, CLOUDSDK_COMPUTE_REGION will be set to this value in mutated containers")
 	gCloudImage := flag.String("gcloud-image", webhooks.GcloudImageDefault, "Container image for the init container setting up GCloud SDK")
+	runAsUser := flag.Int64("run-as-user", webhooks.RunAsUserDefault, "Set the user id for GCloud SDK container")
```

Comment:
I think users can set pod level `securityContext` as below:

```yaml
kind: Pod
spec:
  securityContext:
    runAsUser: 1000
    runAsGroup: 1000
    fsGroup: 1000
  serviceAccountName: keda-operator # assume keda-operator sa has annotations for the webhook
```

Then, gcloud setup container will run as `uid=1000,gid=1000`. Doesn't this help for your use case? Would you mind explaining a bit more detailed reason why the webhook should support this option?

And, I think global and fixed `run-as-user` won't work in several cases. In our on-premise kubernetes cluster, we apply the namespace-per-tenant model. In each namespace, we restrict `uid/gid` which can run pods by PSP. Moreover, the internal webhook will inject security context into pods automatically.

Reply:
I'll try your solution :) 
The case is weird because I tried building a non-root image for gcloud-sdk and even that, I got conflicts with `runAsNonRoot: true`, I'm going to try setting the user at pod level.
Could setting the user-id as annotation be an option in case if this fails?

Reply:
>  I got conflicts with `runAsNonRoot: true`, I'm going to try setting the user at pod level.

I have not actually tried it yet. But, as you pointed out, the default user of gcloud image is root(uid=0).  I think that could conflict with the configuration because kubernetes can't decide which uid should use. So, I expect that setting uid in securityContext will be ok.

> Could setting the user-id as annotation be an option in case if this fails?

Yes, I think so!

Reply:
Okey, I'll try to implement the first option and if I can't, I'll add this feature from annotations.
TBH, IDK why the gcloud sdk needs root... they have literally ignored their own best practices xD

Reply:
> TBH, IDK why the gcloud sdk needs root... they have literally ignored their own best practices xD

I totally agree🙃

Reply:
I have been trying with the securityContext and it works, but I can't set it using kustomize and as we (KEDA) run in OpenShift too, we can't set the user_id more than `runAsNonRoot: true` (and it isn't enough). :( 
I'm going to update the code to get the value from the annotations, if you think it's feasible, it'd be nice, if not, I'll revert the change

Reply:
done :)